src/collector.js

```diff
@@ -22,6 +24,12 @@ export default class Collector {
   }
 
   _addMessage(type, opts, _Message=Message) {
+    if (this.config.scanFile && opts.file) {
+      if (!this.config.scanFile.some(v => v === opts.file)) {
+        return;
+      }
+    }
```

Comment:
Eh, actually, this is simply filtering messages by selected file-name. Still needs comments 😆